### PR TITLE
Import library Linenoise

### PR DESCRIPTION
**Pull request by eecheng87**
Import user-friendly command line library.
Add feature :
1. cursor moving
2. command history
3. auto completion

### CODE REVIEW

**Comment by jserv** [comment]
You must modify `README.md` for external software notice.

**Comment by jserv** [comment]
Also, use `git rebase -i` to rework these git commits.

**Comment by eecheng87** [comment]
> Also, use `git rebase -i` to rework these git commits.

Do you mean I need to delete commit before `7a9ac2c` ?

**Review by jserv — CHANGES_REQUESTED** [review]
Change the section "License" in README.md and mention the licensing terms of linenoise.

**Comment by jserv** [comment]
> > Also, use `git rebase -i` to rework these git commits.
> Do you mean I need to delete commit before `7a9ac2c` ?

Yes, squash and minimize the functional changes.

File: console.c [hunk 1]
```diff
@@ -472,6 +474,7 @@ static void init_in()
     buf_stack = NULL;
 }
 
+//#if 0
```
Comment (jserv) [inline console.c:477]: Remove the dead code.

File: console.c [hunk 2]
```diff
@@ -528,7 +531,9 @@ static char *readline()
 
     return linebuf;
 }
+//#endif
 
+#if 0
```
Comment (jserv) [inline console.c:536]: Remove the dead once you can ensure.

File: qtest.c [hunk 3]
```diff
@@ -34,8 +35,12 @@
 #include "console.h"
 #include "report.h"
 
+
```
Comment (jserv) [inline qtest.c:38]: Don't append blank lines.

File: qtest.c [hunk 4]
```diff
@@ -11,6 +11,7 @@
 #include <sys/wait.h>
 #include <time.h>
 #include <unistd.h>
+#include "dudect/cpucycles.h"
```
Comment (jserv) [inline qtest.c:14]: Why?

**Comment by jserv** [comment]
Use `git rebase` to ensure linenoise specific commits on top of latest `master` branch. Then, force push.

**Comment by eecheng87** [comment]
Can I remove this pull request? I have no idea what I'm doing now... 
I'll open new branch and try to send pull request later.

**Pull request closed without merging**